lib/sisimai/bite/email/verizon.rb

```diff
@@ -0,0 +1,265 @@
+module Sisimai::Bite::Email
+  module Verizon
+    # Sisimai::Bite::Email::Verizon parses a bounce email which created by
+    # Verizon Wireless.
+    # Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/Verizon.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :'received'  => %r/by .+[.]vtext[.]com /,
+        :'vtext.com' => {
+          :'from' => %r/\Apost_master[@]vtext[.]com\z/,
+        },
+        :'vzwpix.com' => {
+          :'from'    => %r/[<]?sysadmin[@].+[.]vzwpix[.]com[>]?\z/,
+          :'subject' => %r/Undeliverable Message/,
+        },
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Verizon Wireless: http://www.verizonwireless.com'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return []; end
+      def pattern
+        return {
+          :from => %r/(?:\Apost_master[@]vtext[.]com|[<]?sysadmin[@].+[.]vzwpix[.]com[>]?)\z/,
+          :subject => Re0[:'vzwpix.com'][:subject],
+        }
+      end
+
+      # Parse bounce messages from Verizon
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match = -1
+        loop do
+          # Check the value of "From" header
+          break unless mhead['received'].find { |a| a =~ Re0[:received] }
+          match = 1 if mhead['from'] =~ Re0[:'vtext.com'][:from]
+          match = 0 if mhead['from'] =~ Re0[:'vzwpix.com'][:from]
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [153.4/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->